build_daemon/lib/src/server.dart

```diff
@@ -74,7 +74,13 @@ class Server {
         if (request is BuildTargetRequest) {
           _buildTargetManager.addBuildTarget(request.target, channel);
         } else if (request is BuildRequest) {
-          await _build(_buildTargetManager.targets, <WatchEvent>[]);
+          var events = <WatchEvent>[];
+          if (request.invalidatedFiles != null) {
+            for (var filePath in request.invalidatedFiles) {
+              events.add(WatchEvent(ChangeType.ADD, filePath));
```

Comment:
We discussed a bit earlier but we will need to figure out what type of change event to provide here.

We might actually need to go investigate all these files and compare them to our last state to create a proper change type?

Reply:
We could also provide the change type ourselves, the flutter tool will know if something is modified/added, though we don't explicitly track that currently

Reply:
I suppose then the protocol would change to add a change type? That would probably be idea from our perspective, as it matches what we need to pass to build runner.